src/components/workspace-utils.js

```diff
@@ -340,6 +341,57 @@ export const WorkspaceTagSelect = props => {
   })
 }
 
+export const WorkspaceStarControl = ({ workspace, starredWorkspaceIds, onUpdate, style }) => {
+  const [updating, setUpdating] = useState(false)
+
+  const { workspace: { workspaceId } } = workspace
+  const isStarred = _.includes(workspaceId, starredWorkspaceIds)
+
+  //Thurloe has a limit of 2048 bytes for its VALUE column. That means we can store a max of 55
+  //workspaceIds in list format. We'll use 50 because it's a nice round number and should be plenty
+  //for the intended use case. If we find that 50 is not enough, consider introducing more powerful
+  //workspace organization functionality like folders
+  const MAX_STARRED_WORKSPACES = 50
+  const maxStarredWorkspacesReached = _.size(starredWorkspaceIds) >= MAX_STARRED_WORKSPACES
+
+  const refreshStarredWorkspacesList = async () => {
+    const { starredWorkspaces } = Utils.kvArrayToObject((await Ajax().User.profile.get()).keyValuePairs)
+    return _.isEmpty(starredWorkspaces) ? [] : _.split(',', starredWorkspaces)
+  }
+
+  const toggleStar = _.flow(
+    Utils.withBusyState(setUpdating),
+    withErrorReporting(`Unable to ${isStarred ? 'unstar' : 'star'} workspace`)
+  )(async star => {
+    const refreshedStarredWorkspaceList = await refreshStarredWorkspacesList()
+    const updatedWorkspaceIds = star ?
+      _.concat(refreshedStarredWorkspaceList, [workspaceId]) :
+      _.without([workspaceId], refreshedStarredWorkspaceList)
+    await Ajax().User.profile.setPreferences({ starredWorkspaces: _.join(',', updatedWorkspaceIds) })
+    authStore.update(_.set('profile.starredWorkspaces', updatedWorkspaceIds))
+    onUpdate(star)
+  })
+
+  return h(Clickable, {
+    as: 'span',
+    role: 'checkbox',
+    'aria-checked': isStarred,
+    tooltip: Utils.cond(
+      [isStarred, () => 'Unstar this workspace'],
+      [!isStarred && !maxStarredWorkspacesReached, () => 'Star this workspace. Starred workspace will appear at the top of your workspace list.'],
```

Comment:
Should be "Starred workspace**s** will appear…"

Reply:
corrected in https://github.com/DataBiosphere/terra-ui/pull/3229/commits/66ea3d50cc4aec80721fb96ee70b6a6d0bf7b671